src/utils.zig

```diff
@@ -408,6 +459,15 @@ pub fn FixedSizeCircularBuffer(T: type, capacity: comptime_int) type { // MARK:
 			if(i >= self.len) return null;
 			return self.mem[(self.startIndex + i) & mask];
 		}
+
+		pub fn isEmpty(self: Self) bool {
+			return self.len == 0;
```

Comment:
Do we really need to abstract this? In my opinion it is nicer to directly see what it's doing instead of having an opaque method for things like this. If used externally you might have a point, but it is only used internally.

Reply:
> Opaque methods are there so you don't need to think about the implementation details after you implement that thing once.

I would generally agree, but here we are in the implementation detail.
And if I do want to read or debug the implementation details of a data structure then I don't want to have to step through 10 different messages just to figure out what's actually going on.